### ISSUE TITLE
Variables accessed before assignment do not follow quoting rules

### ISSUE THREAD

**Issue by InspectorMustache** [swaywm/sway#6723]
Possibly related to #444.

When accessing a variable before its assignment, quotes are not properly removed. For example:

```
bindsym Mod4+1 workspace $ws1

bar {
        position top
}

set $ws1 "a workspace"
```
Pressing Mod4+1 will switch to workspace called `"a workspace"`.

The following will work properly:
```
set $ws1 "a workspace"

bindsym Mod4+1 workspace $ws1

bar {
        position top
}
```
Pressing Mod4+1 with this config will switch to a workspace named `a workspace`.

**Comment by RPigott** [comment]
This is because variables that are not assigned are left unexpanded, and may be expanded later at runtime. It's equivalent to if you had used `$$ws1`. The sway config isn't a shell script, so you have no need for the quotes in the first place. I don't really see what should be changed here.

**Comment by InspectorMustache** [comment]
> The sway config isn't a shell script, so you have no need for the quotes in the first place.

Shouldn't $ws1 always include the quotes then? I think referencing variables in the config file before they're assigned is a bad idea in general so I don't know what should be change either. This just tripped me up when using my i3 config with sway.

**Comment by RPigott** [comment]
Nope. Try `swaymsg workspace this is a test`, works fine and opens a new workspace "this is a test".

I do think it would be sensible to send an error message in swaynag if a config-time replaced var is referenced before assignment, but it really depends on what i3 does there. We should just match whatever they do.
